### PR TITLE
Proper dependency handling for terraform destroy

### PR DESCRIPTION
**Pull request by icereed**
When performing terraform destroy the current code would cause terraform to first delete the tiller setup and then tries to delete the helm releases, which then fails.

By adding explicit and implicit dependecies this is not the case anymore.
It tested it with my Kubernetes cluster on GKE.

Cheers,
icereed

### CODE REVIEW

**Comment by icereed** [comment]
@lsst-sqre do you want me to modify anything in this PR? Please let me know.

File: main.tf [hunk 1]
```diff
@@ -127,6 +127,8 @@ resource "kubernetes_deployment" "tiller_deploy" {
 #}
 
 resource "kubernetes_service_account" "tiller" {
+  depends_on = ["kubernetes_cluster_role_binding.tiller"]
```
Comment (jhoblitt) [inline main.tf]: Isn't this relationship backwards?

**Comment by jhoblitt** [comment]
I apologize for the, ugh, slow response, I apparently wasn't watching this repo.

**Comment by jhoblitt** [comment]
I've cherry-picked the output values commit and merged it as #9

**Comment by jhoblitt** [comment]
I've added some implicit ordering in #10 -- please take a look and let me know if you are still seeing ordering problems.

**Comment by jhoblitt** [comment]
There are unrelated commits on the source branch. I assume that this problem was resolved?

**Pull request closed without merging**